### DISCUSSION TITLE
embedding mp3's in rich-text

### DISCUSSION THREAD

**Discussion by haowens** [tinacms/tinacms#4244] [Ideas]
trying to find a way to make a custom component that can be embedded in the rich-text editor: it would be an audio player that can take in an mp3 from the admin's local file system. Maybe this would requre mp3 support in the media manager or input as a supported content-type?

**Comment by danko-nobre** [comment]
I'm also looking for a feature like this.

**Comment by ussiemer** [comment]
You can write a component that will take the audio src as string and then paste it into your html5 <audio> - should be not that hard to wire it to the media manager too. 

Basic youtube iFrame
`const components = {
    youtube: (props) => {
      return (
        <iframe src={props.youtubeLink} />    
      );
    },
  };`

`<TinaMarkdown content={data.post.body} components={components} />`

and the field template for body:
`templates: [
        {
          name: "youtube",
          label: "Youtube iFrame",
          fields: [
            {
              name: "youtubeLink",
              label: "Youtube Link",
              type: "string",
            }
          ],
        },
      ],`
